include/pybind11/pybind11.h

```diff
@@ -2193,15 +2198,14 @@ class class_ : public detail::generic_type {
             inst, holder_ptr);
     }
 
-    /// Deallocates an instance; via holder, if constructed; otherwise via operator delete.
-    static void dealloc(detail::value_and_holder &v_h) {
-        // We could be deallocating because we are cleaning up after a Python exception.
-        // If so, the Python error indicator will be set. We need to clear that before
-        // running the destructor, in case the destructor code calls more Python.
-        // If we don't, the Python API will exit with an exception, and pybind11 will
-        // throw error_already_set from the C++ destructor which is forbidden and triggers
-        // std::terminate().
-        error_scope scope;
+    // Deallocates an instance; via holder, if constructed; otherwise via operator delete.
+    // We could be deallocating because we are cleaning up after a Python exception.
```

Comment:
We no longer clear the error scope in this function. Consider move the comment.

Reply:
I rewrote the comment, so that this is first:

```
// NOTE: The Python error indicator needs to cleared BEFORE this function is called.
```